ethereum/WebaverseERC721.sol

```diff
@@ -164,94 +171,68 @@ contract WebaverseERC721 is ERC721 {
         require(erc20Contract.transfer(to, amount), "transfer failed");
     }
 
+    function transferFrom(address from, address to, uint256 tokenId) public override {
+        string memory isTransferLocked = getSecureMetadata(tokenId, "isTransferLocked");
+
+        require(
+            keccak256(bytes(isTransferLocked)) != keccak256(bytes("true")),
+            "Cannot transfer when transfer lock is enabled"
+        );
+
+        _transfer(from, to, tokenId);
+    }
+
     /**
-     * @dev Mint one or more non-fungible tokens with this contract
-     * The count parameter is what is looped over to create the token.
-     * This means the hiegher the count, the higher the gas.
-     * This is the main reason that we can only mint so many tokens at once.
+     * @dev Mint one non-fungible token with this contract
      * @param to Address of who is receiving the token on mint
      * Example: 0x08E242bB06D85073e69222aF8273af419d19E4f6
-     * @param hash Hash of the file to mint
-     * Example: 0x1
      * @param name Name of the token
      * @param ext File extension of the token
      * Example: "png"
      * @param description Description of the token (set by user)
-     * @param count Number of tokens to mint (ie: 1)
      */
     function mint(
         address to,
-        string memory hash,
         string memory name,
         string memory ext,
         string memory description,
-        uint256 count
+        uint256 royaltyPercentage,
+        string memory isTransferLocked
```

Comment:
I would make this a bool. It would save a bit of gas and also prevent insane values for this string.